### PR TITLE
Add signal when a professorship is created and deleted ACDM-1144 #resolve

### PR DESCRIPTION
**Pull request by diutsu**

### CODE REVIEW

File: src/main/java/org/fenixedu/academic/domain/Professorship.java [hunk 1]
```diff
@@ -88,6 +94,7 @@ public void delete() {
         ProfessorshipManagementLog.createLog(getExecutionCourse(), Bundle.MESSAGING, "log.executionCourse.professorship.removed",
                 getPerson().getPresentationName(), getExecutionCourse().getNome(), getExecutionCourse()
                         .getDegreePresentationString());
+        ExecutionCourse executionCourse = getExecutionCourse();
```
Comment (Luis-Cruz) [inline src/main/java/org/fenixedu/academic/domain/Professorship.java]: What is this variable for? It is never used in the remainder of the code block.

File: src/main/java/org/fenixedu/academic/domain/Professorship.java [hunk 2]
```diff
@@ -45,6 +47,9 @@
     public static final Comparator<Professorship> COMPARATOR_BY_PERSON_NAME = new BeanComparator("person.name",
             Collator.getInstance());
 
+    public static final String PROFESSORSHIP_CHANGED = "academic.professorship.change";
```
Comment (Luis-Cruz) [inline src/main/java/org/fenixedu/academic/domain/Professorship.java]: Why changed? The Professorship seem to be new.

File: src/main/java/org/fenixedu/academic/domain/Professorship.java [hunk 3]
```diff
@@ -45,6 +47,9 @@
...
+    public static final String PROFESSORSHIP_DELETED = "academic.professorship.deleted";
```
Comment (Luis-Cruz) [inline src/main/java/org/fenixedu/academic/domain/Professorship.java]: This is declared but then it is never used.
Are we missing a Signal.emit() somewhere, or is the constant no necessary?

File: src/main/java/org/fenixedu/academic/domain/Professorship.java (hunk 1 above)
Comment (diutsu) [inline src/main/java/org/fenixedu/academic/domain/Professorship.java]: The idea here was to trigger a signal on deletion, but currently that is not possible (the object will not exist on signal handler invocation). Pull request updated to remove these lines.

**Pull request merged**